t/t2409-sdexec.t

```diff
@@ -0,0 +1,144 @@
+
+test_expect_success 'configure sdexec unit prefix' '
+	cat >systemd.toml <<-EOF &&
+	[systemd]
+	unit-prefix = "flux-t2409-"
+	EOF
+	flux config load <systemd.toml
+'
+test_expect_success 'load sdbus,sdexec modules' '
+	flux exec flux module load sdbus &&
+	flux exec flux module load sdexec
+'
+test_expect_success 'clear broker logs' '
+	flux dmesg -C
+'
+test_expect_success 'sdexec true succeeds' '
+	$sdexec -r 0 true
+'
+test_expect_success 'sdexec false fails with exit code 1' '
+	test_expect_code 1 $sdexec -r 0 false
+'
+test_expect_success 'sdexec /notacmd fails with exit code 127' '
+	test_expect_code 127 $sdexec -r 0 /notacmd
+'
+test_expect_success 'capture broker logs' '
+	flux dmesg >dmesg.log
+'
+test_expect_success 'sdexec stdout works' '
+	cat >hello.exp <<-EOT &&
+	Hello world!
+	EOT
+	$sdexec -r 0 sh -c "echo Hello world!" >hello.out &&
+	test_cmp hello.exp hello.out
+'
+test_expect_success 'sdexec stderr works' '
+	$sdexec -r 0 sh -c "echo Hello world! >&2" 2>hello.err &&
+	test_cmp hello.exp hello.err
+'
+test_expect_success 'sdexec stdout+stderr works' '
+	$sdexec -r 0 sh -c "echo Hello world!; echo Hello world! >&2" \
+		>hello2.out 2>hello2.err &&
+	test_cmp hello.exp hello2.out &&
+	test_cmp hello.exp hello2.err
+'
+test_expect_success 'sdexec stdin works' '
+	echo "Hello world!" | $sdexec -r 0 cat >hello.in &&
+	test_cmp hello.exp hello.in
+'
+test_expect_success 'set up for lptest stdio tests' '
+	$lptest 79 10000 >lptest.exp
+'
+test_expect_success 'sdexec works with 10K lines of stdout' '
+	$sdexec -r 0 cat lptest.exp >lptest.out &&
+	test_cmp lptest.exp lptest.out
+'
+test_expect_success 'sdexec works with 10K lines of stdin' '
+	$sdexec -r 0 sh -c "cat >lptest.in" <lptest.exp &&
+	test_cmp lptest.exp lptest.in
+'
+test_expect_success 'sdexec can set working directory' '
+	pwd >dir.exp &&
+	$sdexec -r 0 pwd >dir.out &&
+	test_cmp dir.exp dir.out
+'
+test_expect_success 'sdexec can set environment' '
+	echo 42 >env.exp &&
+	T2409=42 $sdexec -r 0 printenv T2409 >env.out &&
+	test_cmp env.exp env.out
+'
+# environment modules sets variables with names like 'BASH_FUNC_ml()'
+# https://bugzilla.redhat.com/show_bug.cgi?id=1912046#c5
+test_expect_success 'sdexec ignores environment containing parens' '
+	env "foo()"=badenv666 $sdexec -r 0 printenv >badenv.out &&
+	test_must_fail grep -q badenv666 badenv.out
+'
+
+
+make_exports() {
+	local count=$1
+	local junk=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+	local long=$junk$junk$junk$junk$junk$junk$junk$junk$junk$junk$junk$junk
+	while test $count -gt 0; do
+		echo export TESTVAR_$count=$long
+		count=$(($count-1))
+	done
+}
+
+test_expect_success 'sdexec accepts supersize environment' '
+	count=1024 &&
+	make_exports $count >testenv &&
+	sed -e "s/^export //" <testenv | sort >testenv.exp
+	(. $(pwd)/testenv && $sdexec -r 0 printenv) >testenv.raw &&
+	grep TESTVAR_ testenv.raw | sort >testenv.out &&
+	test_cmp testenv.exp testenv.out
+'
```

Comment:
Test seems a little unexpected.  Is there a bug or something related to this test?  maybe a comment needed here.

Reply:
I inferred somewhere that the dbus server might drop the connection if you send it a message that's too big.  So I was thinking if the user's entire environment is included in a dbus message, there might be some size environment that would trigger a disconnect.  I didn't find any such threshold in experiments, but left the test in just to demonstrate that a pretty big environment works OK.  Maybe a comment to this effect would be helpful.